libs/smart-ngrx/src/selector/virtual-array.class.ts

```diff
@@ -57,5 +57,35 @@ export class VirtualArray<
     });
   }
 
+  /**
+   * Tells the virtual array to refetch the indexes the next
+   * time an element is requested.
+   */
+  refetchIndexes(): void {
+    this.fetchedIndexes = [];
```

Comment:
**suggestion (performance):** Consider implementing granular refetching for better performance

Instead of clearing the entire fetchedIndexes array, consider implementing a more granular approach. You could allow specific indexes or ranges of indexes to be marked for refetching. This could be more efficient in scenarios where only a portion of the data has changed.